R/mergelist.R

```diff
@@ -137,64 +140,79 @@ mergepair = function(lhs, rhs, on, how, mult, lhs.cols=names(lhs), rhs.cols=name
   ans = dtmatch(x=jnto, i=jnfm, on=on, nomatch=nomatch, mult=mult, verbose=verbose, allow.cartesian=allow.cartesian)
 
   ## make i side
-  cp.i = !is.null(ans$irows)
-  out.i = if (!cp.i)
+  out.i = if (is.null(ans$irows))
     .shallow(jnfm, cols=someCols(jnfm, fm.cols, keep=on), retain.key=TRUE)
   else
     .Call(CsubsetDT, jnfm, ans$irows, someCols(jnfm, fm.cols, keep=on))
+  cp.i = cp.i || !is.null(ans$irows)
 
   ## make x side
-  cp.x = !is.null(ans$xrows)
-  out.x = if (!cp.x)
-    .shallow(jnto, cols=someCols(jnto, to.cols, drop=on))
+  out.x = if (is.null(ans$xrows))
+    stop("codecov was here?") #.shallow(jnto, cols=someCols(jnto, to.cols, drop=on)) ## codecov shows we were here but xrows=NULL is not possible as of now, see codecov debug chunk below
   else
     .Call(CsubsetDT, jnto, ans$xrows, someCols(jnto, to.cols, drop=on))
+  cp.x = !is.null(ans$xrows)
+
+  ## debug chunk for codecov to see if `{` makes difference
+  if (FALSE)
+    cat("no!\n")
+  nul = if (FALSE)
+    0L
+  else
+    1L
+  nul = if (FALSE) {
+    0L
+  } else {
+    1L
+  }
```

Comment:
it is indeed an issue in codcov: https://github.com/r-lib/covr/issues/279#issuecomment-622034782

Reply:
added link in comments